luigi/contrib/kubernetes.py

```diff
@@ -247,10 +247,10 @@ def __verify_job_has_started(self):
                         pod.name, t['reason'], t['exitCode'], pod.name)
                     assert t['exitCode'] == 0, err_msg
 
-                assert 'waiting' not in cont_stats['state'],\
-                    "Pod %s %s. Logs: `kubectl logs pod/%s`" % (pod.name,
-                                                                cont_stats['state']['waiting']['reason'],
-                                                                pod.name)
+                if 'waiting' in cont_stats['state']:
+                    wr = cont_stats['state']['waiting']['reason']
+                    assert wr == 'ContainerCreating', "Pod %s %s. Logs: `kubectl logs pod/%s`" % (
```

Comment:
I think you should add the status `Pending` as well to your assert.

Reply:
I have never encountered this case, are you sure that `pod.obj['status'][0]['containerStatuses']['state']['waiting']['reason']` takes `Pending` as value?

Reply:
Nope, I am not sure 🙂 . Gonna trust you there.